Build failure on macOS 10.15

After https://github.com/JuliaLang/julia/pull/49740.  @vtjnash 

```
...
    CC src/ircode.o
    CC src/opaque_closure.o
    CC src/codegen-stubs.o
    CC src/coverage.o
    CC src/runtime_ccall.o
    LINK usr/lib/libjulia-internal.1.10.dylib
Undefined symbols for architecture x86_64:
  "__dyld_dlopen_atfork_parent", referenced from:
      _mach_profile_listener in signal-handling.o
      _signal_listener in signal-handling.o
  "__dyld_dlopen_atfork_prepare", referenced from:
      _mach_profile_listener in signal-handling.o
      _signal_listener in signal-handling.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [/opt/julia/1.10-DEV/julia/usr/lib/libjulia-internal.1.10.dylib] Error 1
make: *** [julia-src-release] Error 2
% sw_vers 
ProductName:	Mac OS X
ProductVersion:	10.15.7
BuildVersion:	19H2026
% pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 12.4.0.0.1.1610135815
volume: /
location: /
install-time: 1683830967
groups: com.apple.FindSystemFiles.pkg-group
```

--- a/src/mach_dyld_atfork.tbd
+++ b/src/mach_dyld_atfork.tbd
@@ -21,5 +21,6 @@ install-name:    '/usr/lib/libSystem.B.dylib'
 exports:
   - targets:         [ arm64-macos, arm64e-macos, x86_64-macos, x86_64-maccatalyst,
                        arm64-maccatalyst, arm64e-maccatalyst ]
-    symbols:         [ __dyld_atfork_parent, __dyld_atfork_prepare ]
+    symbols:         [ __dyld_atfork_parent, __dyld_atfork_prepare,
+                      __dyld_dlopen_atfork_parent, __dyld_dlopen_atfork_prepare ]
 ...

